rootex/script/interpreter.cpp

```diff
@@ -310,6 +325,33 @@ void LuaInterpreter::registerTypes()
 		scene["getID"] = &Scene::getID;
 		scene["getFullName"] = &Scene::getFullName;
 	}
+	{
+		sol::usertype<System> system = rootex.new_usertype<System>("System");
+		system["pause"] = &System::pause;
+		system["unPause"] = &System::unPause;
+
+		sol::table& systems = rootex.create_named("Systems");
+		{
+			sol::usertype<AnimationSystem> animationSystem = systems.new_usertype<AnimationSystem>("Animation", sol::base_classes, sol::bases<System>());
+			sol::usertype<AudioSystem> audioSystem = systems.new_usertype<AudioSystem>("Audio", sol::base_classes, sol::bases<System>());
+			sol::usertype<InputSystem> inputSystem = systems.new_usertype<InputSystem>("Input", sol::base_classes, sol::bases<System>());
+			sol::usertype<LightSystem> lightSystem = systems.new_usertype<LightSystem>("Light", sol::base_classes, sol::bases<System>());
+			sol::usertype<ParticleSystem> particleSystem = systems.new_usertype<ParticleSystem>("Particle", sol::base_classes, sol::bases<System>());
+			sol::usertype<PauseSystem> pauseSystem = systems.new_usertype<PauseSystem>("Pause", sol::base_classes, sol::bases<System>());
+			sol::usertype<PhysicsSystem> physicsSystem = systems.new_usertype<PhysicsSystem>("Physics", sol::base_classes, sol::bases<System>());
+			sol::usertype<PlayerSystem> playerSystem = systems.new_usertype<PlayerSystem>("Player", sol::base_classes, sol::bases<System>());
+			sol::usertype<PostProcessSystem> postProcessSystem = systems.new_usertype<PostProcessSystem>("PostProcess", sol::base_classes, sol::bases<System>());
+			sol::usertype<RenderSystem> renderSystem = systems.new_usertype<RenderSystem>("Render", sol::base_classes, sol::bases<System>());
+			sol::usertype<RenderUISystem> renderUISystem = systems.new_usertype<RenderUISystem>("RenderUI", sol::base_classes, sol::bases<System>());
+			sol::usertype<ScriptSystem> scriptSystem = systems.new_usertype<ScriptSystem>("Script", sol::base_classes, sol::bases<System>());
+			sol::usertype<TransformAnimationSystem> transformAnimationSystem = systems.new_usertype<TransformAnimationSystem>("TransformAnimation", sol::base_classes, sol::bases<System>());
+			sol::usertype<TriggerSystem> triggerSystem = systems.new_usertype<TriggerSystem>("Trigger", sol::base_classes, sol::bases<System>());
+			sol::usertype<UISystem> uiSystem = systems.new_usertype<UISystem>("UI", sol::base_classes, sol::bases<System>());
```

Comment:
these are type definitions, rename them to by of the kind: `sol::usertype<AnimationSystem> animationSystem = systems.new_usertype<AnimationSystem>("AnimationSystem", sol::base_classes, sol::bases<System>());`
and then add property access like this:
```
systems["Animation"] = sol::property(&AnimationSystem::GetSingleton)
```